### PR TITLE
Adding micro-ROS with Teensy and Arduino IDE tutorial

### PR DESCRIPTION
**Pull request by shazib2t**
Adding tutorial for how to connect teensy with micro-ROS and ROS2

### CODE REVIEW

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+ros2 run micro_ros_agent micro_ros_agent serial --dev /dev/ttyACM0
+```
+The result should show something like this:
+
+![Illustration of agent running](micro_ros_agent_start.png)
+
+This means the installation of the micro-ros-agent is successful. 
+Now we can proceed to the next step which is the installation of Arduino IDE 
+and Teensyduino and patching the Arduino-based Teensy board for using the 
+pre-compiled libraries as described [*here*](https://github.com/micro-ROS/micro_ros_arduino#patch-teensyduino).
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:91]:
```suggestion
pre-compiled libraries as described [*in the micro_ros_arduino repository*](https://github.com/micro-ROS/micro_ros_arduino#patch-teensyduino).
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+4. Download the corresponding Teensyduino installer.
+
+5. Run the installer in a termincal by adding execute permission and then execute it.
+$ chmod 755 TeensyduinoInstall.linux64
+$ ./TeensyduinoInstall.linux64
+```
+Now let's set up the patch for the teensy Arduino to use the pre-compiled micro-ros-client 
+libraries, Open a terminal window and follow the commands below: For more information follow 
+the GitHub link from [*micro-ros-arduino*](https://github.com/micro-ROS/micro_ros_arduino/tree/foxy)
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:121]:
```suggestion
the GitHub link from [*micro_ros_arduino*](https://github.com/micro-ROS/micro_ros_arduino/tree/foxy)
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+cd $ARDUINO_PATH/hardware/teensy/avr/
+
+curl https://raw.githubusercontent.com/micro-ROS/micro_ros_arduino/foxy/extras/patching_boards/platform_teensy.txt > platform.txt
+```
+
+Once the above instruction is complete, we will now be able to use the Teensy 3.2 and 
+program it with the pre-compiled micro-ros-client libraries using Arduino IDE.
+
+## Program the Teensy:
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:135]:
```suggestion
## Program the Teensy
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+Once the above instruction is complete, we will now be able to use the Teensy 3.2 and 
+program it with the pre-compiled micro-ros-client libraries using Arduino IDE.
+
+## Program the Teensy: 
+
+Now that we have patched the teensy Arduino IDE, we will be able to use the pre-compiled library by following these instructions:
+
+1. Go to [*link to release section*](https://github.com/micro-ROS/micro_ros_arduino/releases) 
+and download the last release of micro-ROS library for Arduino. 
+Place the file inside ```/home/$USERNAME/Arduino/libraries/ ``` as shown below.
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:141]:
```suggestion
Place the file inside `/home/$USERNAME/Arduino/libraries/` as shown below.
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+![Illustration of patch location](patch_location.png)
+
+Once this process is complete, now let us look at the example folder below:
+
+![Illustration of example location](arduino_example_location.png)
+
+For this tutorial and test, we will be using mico-ros-publisher example as shown above since this 
+program will only publish integer data which will increase in every cycle. Once we selected the 
+example program, we will then upload the code in the teensy 3.2 connected to our host computer
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:151]:
```suggestion
example program, we will then upload the code in the Teensy 3.2 connected to our host computer 
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+![Illustration of example location](arduino_example_location.png)
+
+For this tutorial and test, we will be using mico-ros-publisher example as shown above since this 
+program will only publish integer data which will increase in every cycle. Once we selected the 
+example program, we will then upload the code in the teensy 3.2 connected to our host computer 
+which should show the result as follow.
+
+![Illustration of upload completion](upload_completion.png)
+
+## Running micro-ros-agent in ROS2 Foxy
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:156]:
```suggestion
## Running micro-ROS agent in ROS 2 Foxy
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+Now, let's disconnect the teensy for now from the host computer. We will then open a terminal
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:158]:
```suggestion
Now, let's disconnect the Teensy for now from the host computer. We will then open a terminal 
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+![Illustration of upload completion](upload_completion.png)
+
+## Running micro-ros-agent in ROS2 Foxy
+
+Now, let's disconnect the teensy for now from the host computer. We will then open a terminal 
+or in the docker run the agent program once again as shown at the end of 
+step 2. Make sure to source the ROS path as below:
+
+```bash
+#source /opt/ros/foxy/setup.bash
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:163]:
```suggestion
source /opt/ros/foxy/setup.bash
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+$ source /opt/ros/$ROS_DISTRO/setup.bash
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:164]:
```suggestion
```
$ROS_DISTRO will be set in the setup.bash. Therefore, you have to specify the distro explicitly. As the whole tutorial is based on Foxy, it's fine to write `foxy` directly here.

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+This time we will open another terminal or docker window and type as follow:
+
+```bash
+ros2 topic list
+```
+Which should list as shown below:
+
+![Illustration of ros2 topic connected](ros2_topic_all.png)
+
+See, we have now ```/micro_ros_arduino_node_publisher``` topic publishing in the host computer.
```
Comment (ralph-lange) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:192]:
```suggestion
See, we have now `/micro_ros_arduino_node_publisher` topic publishing in the host computer. 
```

File: _docs/tutorials/core/overview/index.md [hunk 11]
```diff
@@ -25,3 +25,7 @@ If you are new to micro-ROS, we strongly suggest that you take the tutorials in
 * [**Zephyr Emulator**](../zephyr_emulator/)
 
   In this tutorial, you'll learn the use of micro-ROS with Zephyr emulator by testing a Ping Pong application.
+  
+* [**micro-ROS with Teesny using Arduino IDE**](../micro_ROS_with_teensy_arduino/)
```
Comment (ralph-lange) [inline _docs/tutorials/core/overview/index.md:29]: Change name of folder, cf. my second comment.

File: Gemfile.lock [hunk 12]
```diff
@@ -1,13 +1,13 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    activesupport (6.0.3.2)
+    activesupport (6.0.4)
```
[the remark this answers is not in the document]
Comment (shazib2t) [reply Gemfile.lock]: File deleted

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
+---
+title: Micro-ROS with Teensy and Arduino IDE
```
Comment (shazib2t) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:2]:
```suggestion
title: Teensy with Arduino
```

Comment (shazib2t) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:2]:
```suggestion
title: Teensy with Arduino
```

File: _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+permalink: /docs/tutorials/core/micro_ROS_with_teensy_arduino/
```
Comment (shazib2t) [inline _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:3]:
```suggestion
permalink: /docs/tutorials/core/Teensy_with_Arduino/
```

[the remark this answers is not in the document]
Comment (shazib2t) [reply _docs/tutorials/core/micro_ROS_with_teensy_arduino/index.md:3]: Changed below

**Review by shazib2t — COMMENTED** [review]
Modified as suggested. Please check again.

**Review by ralph-lange — APPROVED** [review]

**Comment by ralph-lange** [comment]
Hello @shazib2t, again many thanks for this PR. I had to squash merge it manually due to the deleted Gemfile.lock. (It should have just been removed from the PR, not deleted.) Therefore, your name does not appear directly in the Git history, but I've explicitly added your name to the merge commit message. Best regards, Ralph

**Comment by shazib2t** [comment]
> Hello @shazib2t, again many thanks for this PR. I had to squash merge it manually due to the deleted Gemfile.lock. (It should have just been removed from the PR, not deleted.) Therefore, your name does not appear directly in the Git history, but I've explicitly added your name to the merge commit message. Best regards, Ralph

I accidentally deleted the file but thanks for your understanding and merge. 
Manzur

**Pull request closed without merging**